test/tests/Pathfinding.cpp

```diff
@@ -0,0 +1,249 @@
+        // destination will be picked, to try and get the peep towards the overall pathfinding goal.
+        peep->direction = moveDir;
+        peep->destination_x = peep->x + CoordsDirectionDelta[moveDir].x;
+        peep->destination_y = peep->y + CoordsDirectionDelta[moveDir].y;
+        peep->destination_tolerance = 2;
+
+        // Repeatedly step the peep, until they reach the target position or until the expected number of steps have
+        // elapsed. Each step, check that the tile they are standing on is not marked as forbidden in the test data
+        // (red neon ground type).
+        int step = 0;
+        while (!(*pos == goal) && step < maxSteps)
+        {
+            uint8_t pathingResult = 0;
+            peep->PerformNextAction(pathingResult);
+            ++step;
+
+            pos->x = peep->x / 32;
+            pos->y = peep->y / 32;
+            pos->z = peep->z / 8;
+
+            EXPECT_PRED_FORMAT1(AssertIsNotForbiddenPosition, *pos);
+        }
+        
+        // Clean up the peep, because we're reusing this loaded context for all tests.
+        peep_sprite_remove(peep);
+
+        // Require that the number of steps taken is exactly what we expected. The pathfinder is supposed to be
+        // deterministic, and we reset the RNG seed for each test, everything should be entirely repeatable; as
+        // such a change in the number of steps taken on one of these paths needs to be reviewed.
+        EXPECT_EQ(step, maxSteps);
+
+        return *pos == goal;
+    }
+
+    static ::testing::AssertionResult AssertIsStartPosition(const char*, const TileCoordsXYZ& location)
+    {
+        return AssertPositionIsSetUp("Start", 11u, location);
+    }
+
+    static ::testing::AssertionResult AssertIsGoalPosition(const char*, const TileCoordsXYZ& location)
+    {
+        return AssertPositionIsSetUp("Goal", 9u, location);
+    }
+
+    static ::testing::AssertionResult AssertIsNotForbiddenPosition(const char*, const TileCoordsXYZ& location)
+    {
+        const uint32_t forbiddenSurfaceStyle = 8u;
+
+        const uint32_t style = GetSurfaceStyleAtLocation(TileCoordsXY(location.x, location.y));
+
+        if (style == forbiddenSurfaceStyle)
+            return ::testing::AssertionFailure()
+                << "Path traversed location " << location << ", but it is marked as a forbidden location (surface style "
+                << forbiddenSurfaceStyle << "). Either the map is set up incorrectly, or the pathfinder went the wrong way.";
+
+        return ::testing::AssertionSuccess();
+    }
+
+private:
+
+    static uint32_t GetSurfaceStyleAtLocation(const TileCoordsXY& location)
+    {
+        TileElement* element = map_get_first_element_at(location.x, location.y);
+
+        // Every tile *should* have a surface sprite, so we should be guaranteed to find
+        // something before we go off the end of the data.
+        while (element->GetType() != TILE_ELEMENT_TYPE_SURFACE)
+            element++;
```

Comment:
There is a helper function you can use to do this: `map_get_surface_element_at`

Reply:
Nice. I've eliminated this whole method accordingly. No need for a explicit error checking IMO, it's a highly exceptional circumstance and if it does happen there will be a very obvious access violation to show it.